pulsar-function-go/pf/instance_test.go

```diff
@@ -0,0 +1,93 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package pf
+
+import (
+	"github.com/stretchr/testify/assert"
+	"strconv"
+	"testing"
+	"time"
+)
+
+
+func testProcessSpawnerHealthCheckTimer(tkr *time.Ticker, lastHealthCheckTs int64, expectedHealthCheckInterval int32, counter *int ){
+	println("Starting processSpawnerHealthCheckTimer")
+	now := time.Now()
+	maxIdleTime := int64(time.Duration(expectedHealthCheckInterval) * 3 * time.Second)
+	println("maxIdleTime is: " + strconv.FormatInt(maxIdleTime, 10))
+	timeSinceLastCheck := now.UnixNano() - lastHealthCheckTs
+	println("timeSinceLastCheck is: " + strconv.FormatInt(timeSinceLastCheck, 10))
+	if (timeSinceLastCheck) > (maxIdleTime)  {
+		println("Haven't received health check from spawner in a while. Stopping instance...")
+		// os.Exit(1)
+		tkr.Stop()
+	} else {
+		println("Continuing to check")
+		*counter++
+	}
+}
+
+func testStartScheduler(counter *int){
```

Comment:
```suggestion
func testStartScheduler(counter int){
```

Reply:
Again, this change breaks the test because the counter variable doesn't propagate as needed for the assertion.